### ISSUE TITLE
Importing nonextant private functions from numpy.linalg.linalg in svd_robust

### ISSUE THREAD

**Issue by dehond** [tenpy/tenpy#107]
The svd workaround in `svd_robust.py` imports a few private functions from numpy.linalg.linalg. Several of these have been renamed, and one has even been completely removed in newer numpy versions (I compared 1.13 with 1.18). To wit:

- `_assertRank2` is now `_assert_2d`
- `_assertFinite` is now `_assert_finite`

The content of these two functions in numpy has not changed. However:

- `_assertNoEmpty2d` seems to have been removed altogether. 

Currently, the failure to import these functions stays under the radar if the user has a new enough version of SciPy.

**Comment by jhauschild** [comment]
Good point. Back then, I did this as a hack, assuming that some people don't have access to new scipy versions yet e.g on comuting clusters, where it's also a bit complicated to install new versions.
Did you stumble accross this issue because you also still have a system without a recent enough scipy version?
If not, maybe can drop the support for old scipy versions completely, making this hack obsolete.

**Comment by dehond** [comment]
I ran into the issue because I was using an older version of `svd_robust` at first, where the imports were not put inside a `try`. That worked fine when I tried to use it in Python 2, which I have installed using an older version of numpy (1.13). When I tried to test it with Python 3 I got `NameError`s, because that has numpy 1.18 (with the different function names) installed. 

Dropping support for older scipy versions will make this a non-issue, and I see this has already [been proposed before](https://github.com/tenpy/tenpy/issues/70), so maybe that's more convenient than to implement a new hack to patch the old one.
